amun/api_v1.py

```diff
@@ -273,22 +331,33 @@ def get_inspection_status(inspection_id: str) -> tuple:
     }, 200
 
 
+@versionchanged(
+    version="0.6.0",
+    reason="The function retrieves specification from a Workflow spec."
+)
 def get_inspection_specification(inspection_id: str):
     """Get specification for the given build."""
     parameters = {'inspection_id': inspection_id}
 
     try:
-        build = _OPENSHIFT.get_buildconfig(
-            inspection_id,
-            Configuration.AMUN_INSPECTION_NAMESPACE
+        wf: Dict[str, Any] = _OPENSHIFT.get_workflow(
+            label_selector=f"inspection_id={inspection_id}",
+            namespace=_OPENSHIFT.amun_inspection_namespace
         )
-    except NotFoundException:
+    except NotFoundException as exc:
         return {
-            'error': 'The given inspection id build was not found',
+            'error': 'A Workflow for the given inspection id as not found',
             'parameters': parameters
         }
 
-    specification = json.loads(build['metadata']['annotations']['amun_specification'])
+    parameters: List[Dict[str, Any]] = wf["spec"]["arguments"]["parameters"]
+
+    specification, = map(
+        lambda s: s["value"],
+        filter(lambda p: p["name"] == "specification", parameters)
+    )
```

Comment:
What is this used for?

Reply:
Can you elaborate a bit please?

Reply:
The code is hard to read for me - too much map-lambda-filter.